Lib/fontbakery/profiles/googlefonts.py

```diff
@@ -4374,77 +4375,87 @@ def com_google_fonts_check_repo_zip_files(family_directory):
-        hhea_ascender = ttFont['hhea'].ascent
-        hhea_descender = ttFont['hhea'].descent
+    gf_ttFont = regular_remote_style
+    ttFont = regular_ttFont
 
-        if typo_ascender != expected_ascender:
-            failed = True
-            yield FAIL,\
-                  Message("bad-typo-ascender",
-                          f"{full_font_name}:"
-                          f" OS/2 sTypoAscender is {typo_ascender}"
-                          f" when it should be {expected_ascender}")
+    upm_scale = ttFont['head'].unitsPerEm / gf_ttFont['head'].unitsPerEm
 
-        if typo_descender != expected_descender:
-            failed = True
-            yield FAIL,\
-                  Message("bad-typo-descender",
-                          f"{full_font_name}:"
-                          f" OS/2 sTypoDescender is {typo_descender}"
-                          f" when it should be {expected_descender}")
+    gf_has_typo_metrics = typo_metrics_enabled(gf_ttFont)
+    ttFont_has_typo_metrics = typo_metrics_enabled(ttFont)
 
-        if hhea_ascender != expected_ascender:
+    failed = False
+    if gf_has_typo_metrics:
+        if not ttFont_has_typo_metrics:
             failed = True
-            yield FAIL,\
-                  Message("bad-hhea-ascender",
-                          f"{full_font_name}:"
-                          f" hhea Ascender is {hhea_ascender}"
-                          f" when it should be {expected_ascender}")
+            yield FAIL, Message("bad-fsselection-bit7",
+                                "fsSelection bit 7 needs to be enabled because "
+                                "the family on Google Fonts has it enabled.")
+            # faux enable it so we can see which metrics also need changing
+            ttFont_has_typo_metrics = True
+        expected_ascender = math.ceil(gf_ttFont['OS/2'].sTypoAscender * upm_scale)
+        expected_descender = math.ceil(gf_ttFont['OS/2'].sTypoDescender * upm_scale)
+    else:
+        # if the win metrics have changed, the updated fonts must have bit 7
+        # enabled
+        if (math.ceil(gf_ttFont['OS/2'].usWinAscent * upm_scale),
+            math.ceil(gf_ttFont['OS/2'].usWinDescent * upm_scale)) != \
+           (math.ceil(ttFont['OS/2'].usWinAscent),
+            math.ceil(ttFont['OS/2'].usWinDescent)):
+               if not ttFont_has_typo_metrics:
+                   failed = True
+                   yield FAIL, Message("bad-fsselection-bit7",
+                                       "fsSelection bit 7 needs to be enabled "
+                                       "because the win metrics differ from "
+                                       "the family on Google Fonts.")
+                   ttFont_has_typo_metrics = True
+        expected_ascender = math.ceil(gf_ttFont["OS/2"].usWinAscent * upm_scale)
+        expected_descender = -math.ceil(gf_ttFont["OS/2"].usWinDescent * upm_scale)
+
+    full_font_name = ttFont['name'].getName(4, 3, 1, 1033).toUnicode()
```

Comment:
Our code should be readable even for people who're not like us and do not know these magic numbers by heart ;-)

Reply:
Doh. Apologies.